docs/source/training.rst

```diff
@@ -0,0 +1,276 @@
+        loss.backward()
+        optimizer.step()
+        scheduler.step()
+
+We highly recommend using :func:`~transformers.Trainer`, discussed below, which conveniently
+handles the moving parts of training Transformers models with features like
+mixed precision and easy tensorboard logging.
+
+
+Freezing the encoder
+--------------------
+
+In some cases, you might be interested in keeping the weights of the
+pre-trained encoder frozen and optimizing only the weights of the head
+layers. To do so, simply set the ``requires_grad`` attribute to ``False`` on
+the encoder parameters, which can be accessed with the ``base_model``
+submodule:
+
+.. code-block:: python
+   
+    for param in model.base_model.parameters():
+        param.requires_grad = False
+
+
+.. _tensorflow:
+
+Fine-tuning in native TensorFlow 2
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TensorFlow models can also be trained natively in TF2. Just as with PyTorch,
+TensorFlow models can be instantiated with
+:func:`~transformers.PreTrainedModel.from_pretrained` to load the weights of
+the encoder from a pretrained model. The model can then be trained using the
+Keras ``fit`` method. Here's an example of fine-tuning a sequence classifier
+with a pre-trained encoder on MRPC:
+
+.. code-block:: python
+    
+    import tensorflow as tf
+    import tensorflow_datasets
+    from transformers import TFBertForSequenceClassification, BertTokenizer, glue_convert_examples_to_features
+
+    # Load dataset, tokenizer, model from pretrained model/vocabulary
+    tokenizer = BertTokenizer.from_pretrained('bert-base-cased')
+    model = TFBertForSequenceClassification.from_pretrained('bert-base-cased')
+    data = tensorflow_datasets.load('glue/mrpc')
+
+    # Prepare dataset for GLUE as a tf.data.Dataset instance
+    train_dataset = glue_convert_examples_to_features(data['train'], tokenizer, max_length=128, task='mrpc')
+    valid_dataset = glue_convert_examples_to_features(data['validation'], tokenizer, max_length=128, task='mrpc')
+    train_dataset = train_dataset.shuffle(100).batch(32).repeat(2)
+    valid_dataset = valid_dataset.batch(64)
+
+    # Prepare training: Compile tf.keras model with optimizer, loss and learning rate schedule
+    optimizer = tf.keras.optimizers.Adam(learning_rate=3e-5)
+    loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(optimizer=optimizer, loss=loss)
+
+    # Train and evaluate using tf.keras.Model.fit()
+    history = model.fit(train_dataset, epochs=2, steps_per_epoch=115,
+                        validation_data=valid_dataset, validation_steps=7)
+
+This example uses the built-in
+:func:`~transformers.data.processors.glue.glue_convert_examples_to_features`
+to tokenize MRPC and convert it to a TF ``Dataset`` object which can be
+passed to ``model.fit()``.
+
+With the tight interoperability between TensorFlow and PyTorch models, you
+can even save the model and then reload it as a PyTorch model:
```

Comment:
This is very cool. Does it work backward too? (e.g. can I load a PyTorch-pretrained model back in TF?) Mention the answer either way.

I think this is a cool feature so it should be mentioned in the intro, the reader might not know of it.